chain-abci/src/app/app_init.rs

```diff
@@ -138,20 +141,29 @@ pub struct ChainNodeApp<T: EnclaveProxy> {
     pub accounts: AccountStorage,
     /// valid transactions after DeliverTx before EndBlock/Commit
     pub delivered_txs: Vec<TxAux>,
-    /// root hash of the sparse merkle patricia trie of staking account states after DeliverTx before EndBlock/Commit
-    pub uncommitted_account_root_hash: StarlingFixedKey,
     /// a reference to genesis (used when there is no committed state)
     pub genesis_app_hash: H256,
     /// last two hex digits in chain_id
     pub chain_hex_id: u8,
     /// last application state snapshot (if any)
     pub last_state: Option<ChainNodeState>,
+    /// The state for mempool connection
+    pub mempool_state: Option<ChainNodeState>,
     /// proxy for processing transaction validation requests
     pub tx_validator: T,
     /// was rewards pool updated in the current block?
     pub rewards_pool_updated: bool,
     /// address of tx query enclave to supply to clients (if any)
     pub tx_query_address: Option<String>,
+
+    /// consensus buffer of staking merkle trie storage
+    pub staking_buffer: HashMap<StakedStateAddress, StakedState>,
+    /// mempool buffer of staking merkle trie storage
+    pub mempool_staking_buffer: HashMap<StakedStateAddress, StakedState>,
+    /// consensus buffer of key-value storage
+    pub kv_buffer: HashMap<(u32, Vec<u8>), Option<Vec<u8>>>,
+    /// mempool buffer of key-value storage
+    pub mempool_kv_buffer: HashMap<(u32, Vec<u8>), Option<Vec<u8>>>,
```

Comment:
perhaps create at least type aliases (in chain-storage) for these field types (e.g. `StakingBuffer` and `KVBuffer`).
that way, it'll be a bit clearer in function parameters that it's expecting the chain-storage stuff (instead of potentially some other map) + a bit easier if this is to be switched to something else in the future (e.g. `CHashMap`)

Reply:
Done